packages/csharp/ArmoniK.Api.Worker/Utils/WorkerServer.cs

```diff
@@ -133,10 +133,43 @@ public static WebApplication Create<T>(Action<IServiceCollection, IConfiguration
                                            case GrpcSocketType.Tcp:
+
+                                             var channelOptions = context.Configuration.GetRequiredSection(ComputePlane.SettingSection).
+                                                                          GetSection(ComputePlane.WorkerChannelSection);
+                                             options.Limits.Http2.KeepAlivePingTimeout = ParseTimeSpan(channelOptions["KeepAlivePingTimeout"],
+                                                                                                       TimeSpan.FromSeconds(20));
+                                             options.Limits.KeepAliveTimeout           = ParseTimeSpan(channelOptions["KeepAliveTimeOut"],
+                                                                                                       TimeSpan.FromSeconds(130));
                                              var uri = new Uri(address);
                                              options.ListenAnyIP(uri.Port,
                                                                  listenOptions => listenOptions.Protocols = HttpProtocols.Http2);
                                              break;
+
+                                             /* Local function to parse the KeepAlive options from the configuration section.
+                                              If the section to parse does not exist returns the given defaultValue
+                                              If a valid Time.Span string has been provided it parses and returns it
+                                              Provide support to handle the case where the option has been set to "MaxValue". */
+                                             TimeSpan ParseTimeSpan(string?  toParse,
+                                                                    TimeSpan defaultValue)
+                                             {
+                                               if (string.IsNullOrEmpty(toParse))
+                                               {
+                                                 return defaultValue;
+                                               }
+
+                                               if (toParse.Equals("MaxValue"))
+                                               {
+                                                 return TimeSpan.MaxValue;
+                                               }
+
+                                               if (TimeSpan.TryParse(toParse,
+                                                                     out var result))
+                                               {
+                                                 return result;
+                                               }
+
+                                               throw new FormatException($"Provide a valid TimeSpan format: {toParse} or 'MaxValue'");
+                                             }
```

Comment:
Dependency Injection already does that for you

Reply:
see https://github.com/aneoconsulting/ArmoniK.Core/blob/main/Utils/src/ConfigurationExt.cs

Reply:
Ok, but this does not allow to handle the conversion from "MaxValue" to `TimeSpan.MaxValue` which is the correct value to put to desactivate a given keep alive option, and neither setting a default value. Another point is that the keep alive options might not be processed by the DI at this point, at least is not clear to me ^^